Tests/BackendIntegrationTests/StoreKitIntegrationTests.swift

```diff
@@ -332,6 +332,103 @@ class StoreKit1IntegrationTests: BaseStoreKitIntegrationTests {
 
+    func testRenewalsOnASeparateUserDontTransferPurchases() async throws {
+        let prefix = UUID().uuidString
+        let userID1 = "\(prefix)-user-1"
+        let userID2 = "\(prefix)-user-2"
+
+        let anonymousUser = try self.purchases.appUserID
+        let productIdentifier = try await self.monthlyPackage.storeProduct.productIdentifier
+
+        // 1. Purchase with user 1
+        let user1CustomerInfo = try await self.purchases.logIn(userID1).customerInfo
+        self.assertNoPurchases(user1CustomerInfo)
+        expect(user1CustomerInfo.originalAppUserId) == anonymousUser
+        try await self.purchaseMonthlyOffering()
+
+        // 2. Change to user 2
+        let (identifiedCustomerInfo, _) = try await self.purchases.logIn(userID2)
+        self.assertNoPurchases(identifiedCustomerInfo)
+
+        // 3. Renew subscription
+        self.logger.clearMessages()
+
+        try self.testSession.forceRenewalOfSubscription(productIdentifier: productIdentifier)
+
+        try await self.verifyReceiptIsEventuallyPosted()
+
+        // 4. Verify new user does not have entitlement
+        let currentCustomerInfo = try await self.purchases.customerInfo(fetchPolicy: .fetchCurrent)
+        expect(currentCustomerInfo.originalAppUserId) == userID2
+        self.assertNoPurchases(currentCustomerInfo)
+    }
+
+    func testUserCanMakePurchaseAfterTransferBlocked() async throws {
+        let prefix = UUID().uuidString
+        let userID1 = "\(prefix)-user-1"
+        let userID2 = "\(prefix)-user-2"
+
+        let anonymousUser = try self.purchases.appUserID
+        let productIdentifier = try await self.monthlyPackage.storeProduct.productIdentifier
+
+        // 1. Purchase with user 1
+        var user1CustomerInfo = try await self.purchases.logIn(userID1).customerInfo
+        self.assertNoPurchases(user1CustomerInfo)
+        expect(user1CustomerInfo.originalAppUserId) == anonymousUser
+        try await self.purchaseMonthlyOffering()
+
+        // 2. Change to user 2
+        let (identifiedCustomerInfo, _) = try await self.purchases.logIn(userID2)
+        self.assertNoPurchases(identifiedCustomerInfo)
+
+        // 3. Renew subscription
+        self.logger.clearMessages()
+
+        try self.testSession.forceRenewalOfSubscription(productIdentifier: productIdentifier)
+
+        try await self.verifyReceiptIsEventuallyPosted()
+
+        // 4. Verify new user does not have entitlement
+        var currentCustomerInfo = try await self.purchases.customerInfo(fetchPolicy: .fetchCurrent)
+        expect(currentCustomerInfo.originalAppUserId) == userID2
+        self.assertNoPurchases(currentCustomerInfo)
+
+        // 5. Make purchase with user 2
+        self.logger.clearMessages()
+        try await self.purchaseMonthlyOffering()
+        try await self.verifyReceiptIsEventuallyPosted()
+
+        // 6. Verify user 2 has purchases
+        currentCustomerInfo = try await self.purchases.customerInfo(fetchPolicy: .fetchCurrent)
```

Comment:
This isn't necessary, we could get the `CustomerInfo` returned from the last `self.purchaseMonthlyOffering()` call.

Reply:
Good call, updated! :D